src/command_handler.c

```diff
@@ -0,0 +1,126 @@
+#include "command_handler.h"
+#include "apdu_protocol_defines.h"
+#include "algo_ui.h"
+#include "algo_addr.h"
+#include "algo_keys.h"
+
+
+
+
+void parse_input_get_public_key(const uint8_t* buffer, const uint32_t buffer_len, uint32_t* output_account_id )
+{
+  *output_account_id = 0;
+
+
+  if (buffer_len <= OFFSET_LC) 
+  {
+    PRINTF("using default account id 0 ");
+    return;
+  }
+
+  uint8_t lc = buffer[OFFSET_LC];
+  if (lc == 0)
+  {
+    PRINTF("using default account id 0 ");
+    return ;
+  }
+
+  if (lc < sizeof(uint32_t)) 
+  {
+    THROW(0x6a86);
+  } 
+  
+  if (buffer_len < lc + OFFSET_CDATA)
+  {
+    THROW(0x6a87);
+  } 
+  *output_account_id = U4BE(buffer, OFFSET_CDATA);
+}
+
+
+/*
+* This function takes a binary 32 bytes public key, converts it to Algorand public address,
+* and send it to the UI.
+* this function fails if the public_key buffer is small than 32 bytes.
```

Comment:
nit - `Algorand address` and not `Algorand public address`
typo - small -> smaller